pinot-common/src/main/java/org/apache/pinot/common/cursors/AbstractResponseStore.java

```diff
@@ -0,0 +1,186 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.common.cursors;
+
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.pinot.common.metrics.BrokerMeter;
+import org.apache.pinot.common.metrics.BrokerMetrics;
+import org.apache.pinot.common.response.CursorResponse;
+import org.apache.pinot.common.response.broker.ResultTable;
+import org.apache.pinot.spi.cursors.ResponseSerde;
+import org.apache.pinot.spi.cursors.ResponseStore;
+import org.apache.pinot.spi.env.PinotConfiguration;
+
+
+public abstract class AbstractResponseStore implements ResponseStore {
+
+  protected BrokerMetrics _brokerMetrics;
+
+  /**
+   * Initialize the store.
+   * @param config Configuration of the store.
```

Comment:
In FsResponseStore it looks like the config is already prefixed by something (because there we basically just read properties like `temp.dir` instead of something like `pinot.broker.store.response.fs.temp.dir`). If that is the case, it should specify that here.

If that is not the case... then I guess in `FsResponseStore` we need to use a more specific property name.

Reply:
A subset of the configuration is passed in. I'll add that to the javadoc.